dashboard/src/pages/TreeDetails/TreeDetailsFilter.tsx

```diff
@@ -216,6 +247,42 @@ const CheckboxSection = ({
 
 const MemoizedCheckboxSection = memo(CheckboxSection);
 
+const TimeRangeSection = ({
+  diffFilter,
+  setDiffFilter,
+}: SectionsProps): JSX.Element => {
+  const intl = useIntl();
+
+  const minChangeHandler = useCallback(
+    (e: React.FormEvent<HTMLInputElement>) => {
+      const value = e.currentTarget.value;
+      setDiffFilter(old => ({ ...old, duration_min: parseInt(value) }));
+    },
+    [setDiffFilter],
+  );
+
+  const maxChangeHandler = useCallback(
+    (e: React.FormEvent<HTMLInputElement>) => {
+      const value = e.currentTarget.value;
+      setDiffFilter(old => ({ ...old, duration_max: parseInt(value) }));
+    },
+    [setDiffFilter],
+  );
+
+  return (
+    <DrawerSection>
+      <FilterTimeRangeSection
+        title={intl.formatMessage({ id: 'filter.buildDuration' })}
+        subtitle={intl.formatMessage({ id: 'filter.buildDurationSubtitle' })}
+        min={diffFilter.duration_min}
+        max={diffFilter.duration_max}
+        onMaxChange={maxChangeHandler}
+        onMinChange={minChangeHandler}
+      />
+    </DrawerSection>
```

Comment:
in some cases this section is ready while the others are still loading, it might be a good idea to load them all together